Library/Homebrew/cmd/update.rb

```diff
@@ -228,6 +219,16 @@ def pull!(options = {})
       @initial_branch = ""
     end
 
+    unless `git status --untracked-files=all --porcelain 2>/dev/null`.chomp.empty?
+      if ARGV.verbose?
+        puts "Stashing uncommitted changes to #{repository}."
+        system "git", "status", "--short", "--untracked-files=all"
+      end
+      safe_system "git", "stash", "save", "--include-untracked", *@quiet_args
+      safe_system "git", "reset", "--hard"
```

Comment:
This command always produces output, which might be confusing to most users (that rarely run `brew update` with the `--verbose` option). The output is something along the lines of:

```
HEAD is now at <sha1> <commit-message>
```